src/App.tsx

```diff
@@ -9,39 +10,37 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
 import { User } from './types/User';
 import { Post } from './types/Post';
+import { loadUsers } from './features/usersSlice';
+import { fetchUserPosts } from './features/postsSlice';
+import { RootState, AppDispatch } from './app/store';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useDispatch<AppDispatch>();
+  const users = useSelector((state: RootState) => state.users.items);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
 
-  const [author, setAuthor] = useState<User | null>(null);
+  const posts = useSelector((state: RootState) => state.posts.items);
+  const postsLoaded = useSelector((state: RootState) => state.posts.loaded);
+  const postsError = useSelector((state: RootState) => state.posts.hasError);
   const [selectedPost, setSelectedPost] = useState<Post | null>(null);
```

Comment:
You are using multiple `useSelector` hooks to get different properties of `state.posts`. Consider using a single `useSelector` to get the entire `state.posts` object, and then destructure the properties you need. This can make your code cleaner and more efficient.